Incorrect `main` and `types` Fields in package.json Causes Bundler Resolution Failures

# Incorrect `main` and `types` Fields in package.json Causes Bundler Resolution Failures

## Problem Description

When using `simple-zstd` in a project that uses esbuild (or other bundlers) to bundle code, the module cannot be resolved, causing build failures.

## Error Message

```
✘ [ERROR] Could not resolve "simple-zstd"

    example.ts:7:45:
      7 │ import { ZSTDCompress, ZSTDDecompress } from 'simple-zstd';
        ╵                                              ~~~~~~~~~~~~~

  You can mark the path "simple-zstd" as external to exclude it from the bundle, which will remove
  this error and leave the unresolved path in the bundle.
```

## Root Cause

This is a build configuration mismatch between the TypeScript compiler output and the `package.json` entry points.

The `tsconfig.json` has:
- `outDir: "./dist"`
- Source files are in the `src/` directory

When TypeScript compiles, it preserves the source directory structure, so `src/index.ts` compiles to `dist/src/index.js`.

However, the published `package.json` on npm has:
- `"main": "dist/index.js"`
- `"types": "dist/index.d.ts"`

These paths don't match the actual compiled output location (`dist/src/index.js` and `dist/src/index.d.ts`). Bundlers like esbuild rely on these fields to locate the entry point and TypeScript definitions. When the paths don't match the actual file locations, the module cannot be resolved.

**The fix**: Either set `rootDir: "src"` in `tsconfig.json` to output directly to `dist/` (without the `src/` subdirectory), or update `package.json` to match the current build output structure.

## Expected Solution

There are two possible fixes:

**Option 1: Update package.json to match current build output** (recommended if you want to keep the current TypeScript config):
```json
{
  "main": "dist/src/index.js",
  "types": "dist/src/index.d.ts"
}
```

**Option 2: Update tsconfig.json to output directly to dist/** (requires changing the build config):
```json
{
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "./dist"
  }
}
```
This would make the output go directly to `dist/index.js` (matching the current package.json), but requires recompiling the package.

## Verification

I have verified this issue by:

1. **Reproducing the error**: Running esbuild bundling fails with the module resolution error
2. **Checking the actual file structure**: Confirmed that the compiled files exist at `dist/src/index.js` and `dist/src/index.d.ts`
3. **Applying the fix**: Updated `node_modules/simple-zstd/package.json` to use the correct paths:
   ```json
   "main": "dist/src/index.js",
   "types": "dist/src/index.d.ts"
   ```
4. **Verifying the fix**: After the change, the esbuild bundling process completes successfully without errors

## Request

Please update the published `package.json` on npm to include the correct `main` and `types` fields pointing to the compiled output in the `dist/src/` directory.

I'm not sure which you would rather have

--- a/package.json
+++ b/package.json
@@ -2,8 +2,8 @@
   "name": "simple-zstd",
   "version": "2.0.0",
   "description": "Node.js interface to the system installed zstd.",
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "main": "dist/src/index.js",
+  "types": "dist/src/index.d.ts",
   "engines": {
     "node": ">=20.0.0"
   },
@@ -13,10 +13,10 @@
     "lint": "eslint . --ext .ts",
     "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
     "format:check": "prettier --check \"src/**/*.ts\" \"test/**/*.ts\"",
-    "test": "timeout --signal=TERM --kill-after=5s 120s node --test --require ts-node/register test/test.ts || true",
-    "test-debug": "DEBUG=SimpleZSTD,SimpleZSTDQueue timeout --signal=TERM --kill-after=5s 120s node --test --require ts-node/register test/test.ts || true",
-    "test-ci": "timeout --signal=TERM --kill-after=5s 120s node --test --require ts-node/register test/test.ts && exit 0 || exit 1",
-    "coverage": "timeout --signal=TERM --kill-after=5s 120s node --test --experimental-test-coverage --require ts-node/register test/test.ts || true"
+    "test": "node --test --require ts-node/register test/test.ts || true",
+    "test-debug": "DEBUG=SimpleZSTD,SimpleZSTDQueue timeout -t 120s -- node --test --require ts-node/register test/test.ts || true",
+    "test-ci": "node --test --require ts-node/register test/test.ts && exit 0 || exit 1",
+    "coverage": "node --test --experimental-test-coverage --require ts-node/register test/test.ts || true"
   },
   "repository": {
     "type": "git",

